### PR TITLE
AM-2872 eLinks Consume V2 Data - first commit

### PR DESCRIPTION
**Pull request by mikebrownccd**
https://tools.hmcts.net/jira/browse/AM-2872
eLinks Consume V2 Data

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/reform/orgrolemapping/helper/TestDataBuilder.java [hunk 1]
```diff
@@ -490,6 +517,33 @@ public static Appointment buildAppointmentWithParams(String epimms, String isPri
...
+                .cftRegionID("NULL")
+                .cftRegion("South East")
+                //.locationId("1")
+                //.location("South East")
+                .isPrincipalAppointment(isPrinciple)
+                .appointment(appointment)
+                .appointmentType(appointmentType)
+                .startDate(startDate)
+                .endDate(endDate)
+                //.roles(roles)
```
Comment (mattnayler) [inline src/test/java/uk/gov/hmcts/reform/orgrolemapping/helper/TestDataBuilder.java:543]: Can we update to reflect V2 formats.  e.g.
* remove `courtName`
* populate `cftRegionID`
* remove `locationId` & `locationId`
* remove `roles` **including from input params**

File: src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/ParseRequestServiceTest.java [hunk 2]
```diff
@@ -212,6 +218,17 @@ void validateJudicialProfilesTest() {
                 UserType.JUDICIAL);
     }
 
+    @Test
+    void validateJudicialProfilesTestV2() {
```
[the remark this answers is not in the document]
Comment (mikebrownccd) [reply src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/ParseRequestServiceTest.java:222]: Assertion added

File: src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/RetrieveDataServiceTest.java [hunk 3]
```diff
@@ -236,7 +341,21 @@ void getUserAccessProfile() {
...
+
+    @Test
+    void getUserAccessProfileV2() {
+
+        List<Object> userProfilesResponses = new ArrayList<>();
+        userProfilesResponses.add(TestDataBuilder.buildUserProfilesResponse());
+        ResponseEntity<List<Object>> responseEntity
+                = new ResponseEntity<>(userProfilesResponses, HttpStatus.CREATED);
+
+        Map<String, Set<UserAccessProfile>> response = sutJrdV2.retrieveProfilesByServiceName(responseEntity,
```
[the remark this answers is not in the document]
Comment (mikebrownccd) [reply src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/RetrieveDataServiceTest.java]: Test removed

File: src/test/java/uk/gov/hmcts/reform/orgrolemapping/helper/TestDataBuilder.java (hunk 1 above)
Comment (mikebrownccd) [reply src/test/java/uk/gov/hmcts/reform/orgrolemapping/helper/TestDataBuilder.java]: Recommended changes made

File: src/integrationTest/java/uk/gov/hmcts/reform/orgrolemapping/controller/RefreshControllerIntegrationTest.java [hunk 4]
```diff
@@ -466,6 +468,26 @@ public void shouldProcessRefreshRoleAssignmentsWithJudicialProfiles() throws Exc
         logger.info(" -- Refresh Role Assignment record updated successfully -- ");
     }
 
+    @Test
+    public void shouldProcessRefreshRoleAssignmentsWithJudicialProfilesV2() throws Exception {
+        logger.info(" Refresh role assignments successfully with valid user profiles");
+        var uuid = UUID.randomUUID().toString();
+        doReturn(buildJudicialProfilesResponseV2(uuid)).when(jrdFeignClient).getJudicialDetailsById(any(), any());
```
[the remark this answers is not in the document]
Comment (mikebrownccd) [reply src/integrationTest/java/uk/gov/hmcts/reform/orgrolemapping/controller/RefreshControllerIntegrationTest.java:475]: Agreed to be looked at in another branch, resolved for now as not to hold up testing

File: src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/RetrieveDataServiceTest.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (mattnayler) [reply src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/RetrieveDataServiceTest.java]: Resolved in 53adc1d84ad44a13a4edfda8dc89e71976f97c75.

File: src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/ParseRequestServiceTest.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (mattnayler) [reply src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/ParseRequestServiceTest.java:222]: Resolved in 53adc1d84ad44a13a4edfda8dc89e71976f97c75.

**Review by mattnayler — APPROVED** [review]

**Review by mattnayler — APPROVED** [review]
Approved at f566ac244ceac44ce05254c891378f616e80e602 as suitable to be merged into feature branch: [JRD_eLinks_2_0](https://github.com/hmcts/am-org-role-mapping-service/tree/JRD_eLinks_2_0).

**Review by mattnayler — COMMENTED** [review]
Approved at 98c6a40f4a8468991316453ea5480e8b9dd41e21 as suitable to be merged into feature branch: [JRD_eLinks_2_0](https://github.com/hmcts/am-org-role-mapping-service/tree/JRD_eLinks_2_0).

**Pull request merged**